### ISSUE TITLE
invalid_doc_attributes cannot be allowed or denied

### ISSUE THREAD

**Issue by jyn514** [rust-lang/rust#85497]
I tried this code:

```rust
#![deny(invalid_doc_attributes)]
#![doc(x)]
```

I expected to see this happen: An error that `x` is invalid.

Instead, this happened: The lint stays at a warning and rustc additionally warns that the lint name isn't recognized (https://github.com/rust-lang/rust/pull/85457#issuecomment-844446870):

```
warning: unknown lint: `invalid_doc_attributes`
 --> src/lib.rs:1:9
  |
1 | #![deny(invalid_doc_attributes)]
  |         ^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unknown_lints)]` on by default

warning: unknown `doc` attribute `x`
 --> src/lib.rs:2:8
  |
2 | #![doc(x)]
  |        ^
  |
  = note: `#[warn(invalid_doc_attributes)]` on by default
  = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
  = note: for more information, see issue #82730 <https://github.com/rust-lang/rust/issues/82730>
```

### Meta

`rustc --version --verbose`:  1.54.0-nightly (2021-05-18 4e3e6db011c5b482d2be)

**Comment by jyn514** [comment]
Strangely, `#![deny(forbidden_lint_groups)]` (another future incompat lint) also has no effect, but doesn't give a warning that it's unknown.
